shark_turbine/kernel/compiler/kernel_codegen.py

```diff
@@ -191,19 +205,31 @@ def add_grid(self, grid: Type[Grid]):
             )
 
     def determine_input_output_buffers(self, graph: fx.Graph):
-        placeholder_nodes: list[fx.Node] = []
-        for node in graph.nodes:
-            if node.op != "placeholder":
-                continue
-            placeholder_nodes.append(node)
+        # Extract all placeholder nodes.
+        def is_placeholder(node: fx.Node):
```

Comment:
is this function duplicated?

Reply:
Yes, sorry, I am still not sure on the decorum on how/where to put these util/helper function. At first I was thinking of using it similar to how we use lambda functions in C++. 

De-duplicated though, LMK if you think there is a more apt spot. :)